File: varuna-backend/app/api/v1/notifications.py
"""
Notifications API Endpoints
"""

from typing import List
from fastapi import APIRouter, Body
from app.schemas.notifications import NotificationSchema, RegisterDeviceRequest
from app.services.notification_service import notification_service

router = APIRouter()


@router.get("", response_model=List[NotificationSchema])
def get_notifications():
    return notification_service.get_notifications()


@router.post("/register-device")
def register_device(payload: RegisterDeviceRequest = Body(...)):
    return {"success": True, "token": payload.push_token}


@router.post("/{notification_id}/read")
def mark_read(notification_id: str):
    return {"success": True, "id": notification_id}

File: Varuna-Frontend/src/data/api/notifications.ts
/**
 * VARUNA Notifications API
 */

import { apiClient } from './client';
import { AppNotification } from '../../domain/models/types';

export const notificationsApi = {
  async getNotifications(limit: number = 20): Promise<AppNotification[]> {
    return apiClient.get<AppNotification[]>('/notifications', {
      params: { limit },
    });
  },

  async registerPushToken(token: string, platform: string): Promise<{ success: boolean }> {
    return apiClient.post<{ success: boolean }>('/notifications/register-device', {
      push_token: token,
      platform,
    });
  },

  async markAsRead(notificationId: string): Promise<{ success: boolean }> {
    return apiClient.post<{ success: boolean }>(`/notifications/${notificationId}/read`);
  },
};
